fairworkflows/fairstep.py

```diff
@@ -93,6 +94,11 @@ def __init__(self, label: str = None, description: str = None, uri=None,
             self.inputs = inputs
         if outputs is not None:
             self.outputs = outputs
+        # Set temporary URI to refer to this step in workflows
+        # (i.e. 'http://fairworkflows.org#8769029329049')
+        if uri is None:
+            self._uri = DUMMY_FAIRWORKFLOWS_URI + '#step' + str(hash(self))
```

Comment:
I like this, at least, it's better than what we were doing before for unpublished workflows.